test/e2e/specs/osba_app.go

```diff
@@ -0,0 +1,129 @@
+package specs
+
+import (
+	"context"
+	"io/ioutil"
+	"os"
+	"time"
+
+	. "github.com/onsi/ginkgo"
+	. "github.com/onsi/gomega"
+
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/util/wait"
+
+	"github.com/openshift/openshift-azure/pkg/util/ready"
+	"github.com/openshift/openshift-azure/test/e2e/standard"
+	"github.com/openshift/openshift-azure/test/sanity"
+)
+
+type osbaObj struct {
+	namespace string
+	data      []byte
+}
+
+var osbaApp, osbaService osbaObj
+
+func init() {
+	osbaApp.namespace = "osba"
+	osbaService.namespace = "osba"
+
+	var err error
+	osbaService.data, err = ioutil.ReadFile("../manifests/osba/osba-os-template.yaml") // running via `go test`
+	if os.IsNotExist(err) {
+		osbaService.data, err = ioutil.ReadFile("test/e2e/manifests/osba/osba-os-template.yaml") // running via compiled test binary
+	}
+	if err != nil {
+		panic(err)
+	}
+}
+
+// ValidateOSBAApp verifies that the created application runs normally
+func validateOSBAApp(ctx context.Context, cookie interface{}) []*standard.TestError {
+	var errs []*standard.TestError
+
+	sanity.Checker.Log.Debugf("validating that all osba components are healthy")
+	err := wait.Poll(2*time.Second, 20*time.Minute, ready.CheckClusterServiceBrokerIsReady(sanity.Checker.Client.CustomerAdmin.ServicecatalogV1beta1.ClusterServiceBrokers(), osbaService.namespace))
+	if err != nil {
+		sanity.Checker.Log.Error(err)
+		errs = append(errs, &standard.TestError{Err: err, Bucket: "ready.CheckClusterServiceBrokerIsReady"})
```

Comment:
(the point of the buckets is that they are few - "validateOSBAApp" or even just "osba" is better - but prefer to return error)